README.md

```diff
@@ -100,41 +121,50 @@ Each line is a single event collected by Tracee, with the following information:
 7. EVENT - identifies the event (e.g. syscall name)
 8. ARGS - list of arguments given to the function
 
-When using table-verbose output, the following information is added:
+### Customizing the output
+
+Tracee supports different output formats. For example, to choose json output, use `--output json`.
+
+To tell tracee to write events to a file instead of stdout, use `--output out-file:/path/to/file`.
+
+There are different ways you can augment the output to add useful information. For example: `--output eot` will add a terminating event to the stream which is useful if feeding Tracee's output to another program.
+
+For a full list of output options, run `--output help`.
+
+### Selecting what to trace
+
+Trace output can easily become unwieldily when tracing all of the events from all of the system. Luckily, Tracee has a powerful mechanism to accurately trace just the information that is relevant to you, using the `--trace` flag.
+Using the `--trace` you define expressions that tells Tracee what you are interested in by means of event metadata, and process metadata. Only events that match this criteria will be traced.
+
+You can filter by most of the visible fields on a Tracee event. For example to trace only events related to user ID 1000, use `--trace uid=1000`.  
+You can combine trace expressions into more complex criteria. For example, to trace only events related to user ID 1000, which come from process ID 1234, use `--trace uid=1000 --trace pid=1234`.  
+
+A special `pid` value is `new` which let's you trace all newly created processes (that were created after Tracee started tracing).  
+Tracee lets you easily trace events that originate in containers using `--trace container` or only new containers (that was created after Tracee started) using `--trace container=new`.
+
+Event metadata can be used in trace expression as well. For example, to trace only `openat` syscalls, use `--trace event:openat`. But you can also filter on a specific argument of the event, e.g `--trace openat.pathname=/bin/ls` which will show only `openat` syscalls that operate on the file `/bin/ls`.
+
+A useful trace mode is the `--trace follow` which, if specified, will trace not only processes that match the given trace expressions, but also their child processes.
+For example, the following will trace all the events that originate from zsh shell, including all of the processes that it will spawn: `--trace command=zsh --follow`.
 
-1. UTS_NAME - uts namespace name. As there is no container id object in the kernel, and docker/k8s will usually set this to the container id, we use this field to distinguish between containers.
-2. MNT_NS - mount namespace inode number.
-3. PID_NS - pid namespace inode number. In order to know if there are different containers in the same pid namespace (e.g. in a k8s pod), it is possible to check this value
-4. PPID - parent pid of the calling process
+For a complete list of trace options, run `--trace help`.
```

Comment:
When I tried to see the help today, I instinctively passed it as the following:
```
./tracee --trace --help
2021/01/22 11:34:22 invalid filter option specified, use '--filter help' for more info
```

I realize we currently only look at the arg to have "help" and not "--help": https://github.com/aquasecurity/tracee/blob/main/main.go#L345-L347

Should we add "--help" as well?

Reply:
Should we return an error (invalid filter) in this case: `./tracee --trace 'uid<0'` same for `./tracee --trace 'pid<0'`? currently we start tracing

Reply:
What is expected to happen with `--trace '!c'`? I currently don't see we handle it here: https://github.com/aquasecurity/tracee/blob/main/main.go#L440 but if `--trace '!container'` and `--trace c` are a thing, intuitively I think `--trace '!c'` would be too.

Reply:
Doing --trace '!c' works exactly as you expect. Give it a try (and then you can also see why the hasprefix in line 440 works - it's 'f' being the prefix of "!container")

Reply:
Ah yes - I missed that detail. It does get caught by '!c'. I think we should add it to the readme as an example.

Reply:
Regarding pid/uid<0 - I agree. We should produce an error when such values are given.
Also agree about the --help thing (and -h as well).
These two, of course, are for different commits.

Regarding the '!c' in the readme - it previously was an example in the --trace help, but I removed it as I already put there "container", "c" and "!container", and I thought it was only logical that people will understand what "!c" does.

Reply:
Roger - I created an issue to track it: https://github.com/aquasecurity/tracee/issues/450